starfish/util/logging.py

```diff
@@ -0,0 +1,51 @@
+import platform
+from functools import lru_cache
+from json import JSONEncoder
+from subprocess import (
+    CalledProcessError,
+    check_output,
+)
+
+import pkg_resources
+
+import starfish
+from starfish.types import CORE_DEPENDENCIES
+
+
+@lru_cache(maxsize=1)
+def get_core_dependency_info():
+    dependency_info = dict()
+    for dependency in CORE_DEPENDENCIES:
+        version = pkg_resources.get_distribution(dependency).version
+        dependency_info[dependency] = version
+    return dependency_info
+
+
+@lru_cache(maxsize=1)
+def get_git_commit_hash():
```

Comment:
this method, as written, is subject to strange working directory effects.  i suspect you should run the commands in the parent path of `starfish.__file__`.

Reply:
starfish.__file__ outputs the file directory path though so does it matter?